lib/modules/keyboardNav.js

```diff
@@ -158,51 +159,71 @@ const commands = {
 		include: ['comments', 'inbox'],
 		value: [75, false, false, false], // k
 		description: 'Move up to the previous comment on threaded comment pages',
+		callback() { move(thing => thing.getNext({ direction: 'up' })); },
 	},
 	moveDownComment: {
 		include: ['comments', 'inbox'],
 		value: [74, false, false, false], // j
 		description: 'Move down to the next comment on threaded comment pages',
+		callback() { move(thing => thing.getNext({ direction: 'down' })); },
 	},
 	moveTop: {
 		include: ['linklist', 'modqueue', 'profile', 'inbox', 'search'],
 		value: [75, false, false, true], // shift-k
 		description: 'Move to top of list (on link pages)',
+		callback() { move(Thing.visibleThingElements()[0], { scrollStyle: null }); },
 	},
 	moveBottom: {
 		include: ['linklist', 'modqueue', 'profile', 'inbox', 'search'],
 		value: [74, false, false, true], // shift-j
 		description: 'Move to bottom of list (on link pages)',
+		callback() { move(Thing.visibleThingElements().slice(-1)[0], { scrollStyle: 'top' }); },
 	},
 	moveUpSibling: {
 		include: ['comments'],
 		value: [75, false, false, true], // shift-k
 		description: 'Move to previous sibling (in comments) - skips to previous sibling at the same depth.',
+		callback() {
+			move(thing =>
+				thing.getNextSibling({ direction: 'up' }) || thing.getNext({ direction: 'up' }),
+				{ scrollStyle: 'legacy' }
+			);
+		},
 	},
 	moveDownSibling: {
 		include: ['comments'],
 		value: [74, false, false, true], // shift-j
 		description: 'Move to next sibling (in comments) - skips to next sibling at the same depth.',
+		callback() {
+			move(thing =>
+				thing.getNextSibling({ direction: 'down' }) || thing.getNext({ direction: 'down' }),
+				{ scrollStyle: 'legacy' }
+			);
+		},
 	},
 	moveUpThread: {
 		include: ['comments'],
 		value: [75, true, false, true], // shift-alt-k
 		description: 'Move to the topmost comment of the previous thread (in comments).',
+		callback() { move(thing => thing.getThreadTop().getNextSibling({ direction: 'up' })); },
```

Comment:
We need `scrollStyle: 'legacy'` for this and `moveDownThread` (looks like that's what it was before?)